### PR TITLE
♻️(widgets) refacto of widgets layout

### PR DESCRIPTION
**Pull request by Kitero**
## Purpose

Refacto of widgets layout to match Richie project standards

### CODE REVIEW

File: src/types/Widgets.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { JSONSchema7TypeName } from 'json-schema';
+import type { WidgetProps } from '@rjsf/core';
+import { ChangeEventHandler } from 'react';
 
 export default interface ShowCaseWidgetProps {
   widgetName: string;
   type: JSONSchema7TypeName;
 }
+
+export type VernaWidgetProps = Partial<WidgetProps> & { onChange: ChangeEventHandler<unknown> };
```
Comment (jbpenrath) [inline src/types/Widgets.ts:10]: Just for homogeneity to define Props type we could use `interface` or `type` but if possible not both.

```suggestion
export interface VernaWidgetProps extends Partial<WidgetProps> {
  onChange: WidgetProps['onChange'] & ChangeEventHandler<unknown>;
};
```

Furthermore, just to be sure, do you really want the type of `onChange` is `WidgetProps['onChange'] & ChangeEventHandler<unknown>` or do you want that is equal to `ChangeEventHandler<unknown>` ?

File: examples/playground/src/widgets/templates/Field.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}: PropsWithChildren<FieldProps>) {
+  const classList = useMemo(() => {
+    const classes = ['form-field', ...fieldClasses];
+    if (error) {
+      classes.push('form-field--error');
+    }
+    return classes.join(' ');
+  }, [fieldClasses, error]);
+
+  const iconHref = error ? '#icon-warning' : '#icon-info-rounded';
```
Comment (jbpenrath) [inline examples/playground/src/widgets/templates/Field.tsx:23]: Those icons does not exist here.

File: src/types/Widgets.ts (hunk 1 above)
Comment (Kitero) [reply src/types/Widgets.ts:10]: I guess I wanted the onchange of ChangeEventHandler<unknown> and not the one of widgetProps, but i now have a warning that event.target.value doesn't exist.. (in both cases, weird)

File: examples/playground/src/widgets/templates/Field.tsx (hunk 2 above)
Comment (Kitero) [reply examples/playground/src/widgets/templates/Field.tsx:23]: Indeed

File: src/types/Widgets.ts (hunk 1 above)
Comment (Kitero) [reply src/types/Widgets.ts]: I overloaded a onChange on the input template and it's good now

File: examples/playground/src/widgets/CheckboxWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export default function CheckboxWidget({
+  disabled,
+  rawErrors,
+  onChange,
+  id,
+  label,
+  required,
+  value,
+}: VernaWidgetProps) {
+  const [checked, setChecked] = useState<boolean>(value || false);
```
Comment (jbpenrath) [inline examples/playground/src/widgets/CheckboxWidget.tsx:14]: So `value` prop is not updated when we call `onChange` ?

File: examples/playground/src/widgets/TextareaWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  onChange,
+  value,
+}: VernaWidgetProps) {
+  return (
+    <Field error={error} message={message}>
+      <textarea
+        aria-label={label}
+        className="form-field__textarea"
+        disabled={disabled}
+        id={id || ''}
```
[the remark this answers is not in the document]
Comment (jbpenrath) [reply examples/playground/src/widgets/TextareaWidget.tsx]: Apply this comment to all widget

File: examples/playground/src/widgets/templates/TextTemplateWidget.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import Field, { FieldProps } from './Field';
+import { ChangeEventHandler, InputHTMLAttributes } from 'react';
+
+interface TextTemplateWidgetProps extends InputHTMLAttributes<unknown>, FieldProps {
+  id: string;
```
Comment (jbpenrath) [inline examples/playground/src/widgets/templates/TextTemplateWidget.tsx:5]: Let's use the `id` as it is typed by `InputHTMLAttributes`
```suggestion
```

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**